worker/runtime/integration/integration_test.go

```diff
@@ -434,6 +434,65 @@ func (s *IntegrationSuite) TestContainerAllowsHostAccess() {
 	s.Equal(exitCode, 0, "Process in container should be able to reach the host network")
 }
 
+func (s *IntegrationSuite) TestContainerNetworkHosts() {
+	namespace := "test-restricted-networks"
+	requestTimeout := 3 * time.Second
+
+	network, err := runtime.NewCNINetwork()
+
+	s.NoError(err)
+
+	networkOpt := runtime.WithNetwork(network)
+	customBackend, err := runtime.NewGardenBackend(
+		libcontainerd.New(
+			s.containerdSocket(),
+			namespace,
+			requestTimeout,
+		),
+		networkOpt,
+	)
+	s.NoError(err)
+
+	s.NoError(customBackend.Start())
+
```

Comment:
We don't need to create a custom garden backend if we're not changing anything from the default values. Use `s.gardenBackend` instead.
```suggestion
```